File: tests/indexing/parsing/lang/fixtures/java/Loops.java
class Loops {

    String join(String[] parts) {
        StringBuilder out = new StringBuilder();
        for (int i = 0; i < parts.length; i++) {
            out.append(parts[i]);
            out.append(",");
        }
        return out.toString();
    }

    int largest(int[] xs) {
        int max = xs[0];
        for (int x : xs) {
            max = x > max ? x : max;
        }
        return max;
    }

    long fib(int n) {
        long a = 0;
        long b = 1;
        while (n > 0) {
            long next = a + b;
            a = b;
            b = next;
            n = n - 1;
        }
        return a;
    }

    int digits(int n) {
        int count = 0;
        do {
            n = n / 10;
            count = count + 1;
        } while (n != 0);
        return count;
    }
}

File: tests/indexing/parsing/lang/fixtures/kotlin/Loops.kt
class Loops {

    fun join(parts: List<String>): String {
        val out = StringBuilder()
        for (part in parts) {
            out.append(part)
            out.append(",")
        }
        return out.toString()
    }

    fun fib(n: Int): Long {
        var a = 0L
        var b = 1L
        var count = n
        while (count > 0) {
            val next = a + b
            a = b
            b = next
            count -= 1
        }
        return a
    }

    fun digits(n: Int): Int {
        var value = n
        var count = 0
        do {
            value /= 10
            count += 1
        } while (value != 0)
        return count
    }
}
